File: 프로그래머스/1/42862. 체육복/체육복.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool l[35], r[35];
int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = n - lost.size();
    for(auto x: lost) l[x] = 1;
    for(auto x: reserve) r[x] = 1;
    for(int i = 1; i <= n; i++) {
        if(l[i] && r[i]) {
            l[i] = r[i] = 0;
            answer++;
        }
    }
    for(int i = 1; i <= n; i++){
        if(l[i]) {
            if(r[i-1]) {
                answer++;
                r[i-1] = 0;
            }
            else if(r[i+1]){
                answer++;
                r[i+1] = 0;
            }
        }
    }
    
    return answer;
}

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = n - lost.length;
        boolean[] l = new boolean[n + 2];
        boolean[] r = new boolean[n + 2];
        
        for(int x : lost)
            l[x] = true;
        for(int x : reserve)
            r[x] = true;
        
        for(int i = 1; i <= n; i++) {
            if(l[i] && r[i]) {
                l[i] = r[i] = false;
                answer++;
            }
        }
        
        for(int i = 1; i <= n; i++){
            if(!l[i]) continue;
            if(r[i - 1]) {
                r[i - 1] = false;
                answer++;
            } else if(r[i + 1]) {
                r[i + 1] = false;
                answer++;
            }
        }
        
        
        return answer;
    }
}